api/src/main/java/org/openmrs/obs/handler/AbstractHandler.java

```diff
@@ -59,46 +55,31 @@ public AbstractHandler() {
 	 * @return File that the complex data should be written to
 	 */
 	public File getOutputFileToWrite(Obs obs) throws IOException {
-		// Get the title and remove the extension.
+		// Get the title and the extension
 		String title = obs.getComplexData().getTitle();
-		String extension = "." + getExtension(title);
-		
-		// If getExtension returns the title, there was no extension
-		if (getExtension(title).equals(title)) {
-			extension = "";
-		}
+		String extension = FilenameUtils.getExtension(title);
 		
-		File dir = OpenmrsUtil.getDirectoryInApplicationDataDirectory(Context.getAdministrationService().getGlobalProperty(
-		    OpenmrsConstants.GLOBAL_PROPERTY_COMPLEX_OBS_DIR));
-		File outputfile;
+		File dir = OpenmrsUtil.getDirectoryInApplicationDataDirectory(
+		    Context.getAdministrationService().getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_COMPLEX_OBS_DIR));
+		File outputfile = null;
 		
-		// Get the output stream
+		// Generate the filename with the Obs' UUID
+		String uuid = obs.getUuid();
```

Comment:
@FloBue nice work! I love it when we remove so much code and deprecate methods :)

I have a suggestion I find a bit cleaner, what do you say?

```java
File dir = ...
String uuid = obs.getUuid();
// should also catch title being null since getExtension(null) returns null but `defaultIfEmpty(null, "dat")` returns "dat"
String extension = "." + StringUtils.defaultIfEmpty(FilenameUtils.getExtension(title), "dat");
String filename;
String titleWitoutExtension = FilenameUtils.removeExtension(title);
if (StringUtils.isNotBlank(titleWitoutExtension) {
  filename =  titleWitoutExtension + "_" + uuid + extension;
} else {
  filename =  uuid + extension;
}
File outputfile = new File(dir, filename);
```

Reply:
one thing I dont know which concerns your/my implementation is: can a title be like "a\b\c.jpg" ? removeExtension says it makes "a\b\c.jpg  --> a\b\c" but there is also `getBaseName` which handles such cases differently